### PR TITLE
feat: scrubbing preview

### PR DESCRIPTION
**Pull request by Faywyrr**
Add scrubbing preview with Jellyscrub plugin

Demo :
https://user-images.githubusercontent.com/51246506/219468275-22c4d8b6-b16e-49bd-9ab9-263604c991ed.mp4

### CODE REVIEW

**Comment by jarnedemeulemeester** [comment]
Nice work!

**Comment by Faywyrr** [comment]
@OpenSourceSimon look at the last commit

**Comment by OpenSourceSimon** [comment]
> @OpenSourceSimon look at the last commit

Awesome thanks!

**Review by jarnedemeulemeester — CHANGES_REQUESTED** [review]
Overall pretty happy with the implementation. It's very smooth and looks good. 
I didn't encounter any bugs :smile: 

I do however have some proposed changes. 

I have a commit ready, if you agree with the changes I will push it.

**Comment by jarnedemeulemeester** [comment]
Seems like one of my comments went missing... 

I also would move the retrieval of the trick play data to `onMediaItemTransition`. That way it only retrieves it for the current item and does it async. 
For example right now if there are 80 items queued with each 2MB of trick play data then the player needs to do 160 extra requests for a total of 160MB of data. This could block the player for quite some time.

But as I've mentioned I have a commit ready :smile:

File: app/phone/src/main/res/layout/exo_player_control_view.xml [hunk 1]
```diff
@@ -173,6 +173,22 @@
...
         android:padding="16dp">
 
+        <FrameLayout
+            android:id="@+id/preview_frame_layout"
+            android:layout_width="160dp"
+            android:layout_height="90dp"
+            android:layout_gravity="start"
+            android:background="@android:color/transparent"
+            android:visibility="gone"
+            tools:visibility="visible">
```
[the remark this answers is not in the document]
Comment (Faywyrr) [reply app/phone/src/main/res/layout/exo_player_control_view.xml:183]: So much the better if it works without the framelayout

File: app/phone/src/main/java/dev/jdtech/jellyfin/utils/PreviewScrubListener.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        val currentBifData = currentTrickPlay.value ?: return
+        val image = BifUtil.getTrickPlayFrame(position.toInt(), currentBifData) ?: return
+
+        val parent = previewFrameLayout.parent as ViewGroup
+        val layoutParams = previewFrameLayout.layoutParams as ViewGroup.MarginLayoutParams
+
+        val offset = position.toFloat() / player.duration
+        val minX = previewFrameLayout.left
+        val maxX = parent.width - parent.paddingRight - layoutParams.rightMargin
```
[the remark this answers is not in the document]
Comment (Faywyrr) [reply app/phone/src/main/java/dev/jdtech/jellyfin/utils/PreviewScrubListener.kt:46]: so you might as well take it off :)

File: app/phone/src/main/java/dev/jdtech/jellyfin/utils/PreviewScrubListener.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import dev.jdtech.jellyfin.utils.bif.BifData
+import dev.jdtech.jellyfin.utils.bif.BifUtil
+import timber.log.Timber
+
+class PreviewScrubListener(
+    private val previewFrameLayout: View,
+    private val scrubbingPreview: ImageView,
+    private val timeBarView: View,
+    private val player: Player,
+    private val currentTrickPlay: LiveData<BifData?>
```
[the remark this answers is not in the document]
Comment (Faywyrr) [reply app/phone/src/main/java/dev/jdtech/jellyfin/utils/PreviewScrubListener.kt:20]: 👍

**Comment by Faywyrr** [comment]
@jarnedemeulemeester Yes I suspected that loading all the previews in cache would surely be a bit problematic. For the rest it's great if it works as well or better, I let you commit it. Thanks for your speed!

**Review by jarnedemeulemeester — APPROVED** [review]

**Comment by jarnedemeulemeester** [comment]
@Faywyrr I recommend you test again to make sure everything still works.

If it's all good then I think we can merge this!

**Comment by Faywyrr** [comment]
@jarnedemeulemeester on my side it works fine !

**Pull request merged**